percona_pg_telemetry.c

```diff
@@ -780,12 +826,153 @@ get_extensions_list(PTDatabaseInfo *dbinfo, MemoryContext cxt)
 	return extlist;
 }
 
+static List *
+get_access_methods_list(MemoryContext cxt)
+{
+	List	   *amlist = NIL;
+	SPITupleTable *tuptable;
+	int			spi_result;
+	char	   *query = "SELECT am.amname AS access_method, count(*) FROM pg_class c LEFT JOIN pg_am am ON c.relam = am.oid WHERE c.relkind = 'r' AND c.relnamespace NOT IN ( SELECT oid FROM pg_namespace WHERE nspname IN ('pg_catalog', 'information_schema', 'pg_toast') ) GROUP BY access_method ORDER BY access_method;";
+	MemoryContext oldcxt;
+
+	SetCurrentStatementStartTimestamp();
+	StartTransactionCommand();
+
+	/* Initialize SPI */
+	if (SPI_connect() != SPI_OK_CONNECT)
+	{
+		ereport(ERROR, (errmsg("Failed to connect to SPI")));
+	}
```

Comment:
You can removed this code ince we do not support PostgreSQL 10.

> The fact that these functions return int not void is historical. All failure cases are reported via ereport or elog. (In versions before PostgreSQL v10, some but not all failures would be reported with a result value of SPI_ERROR_CONNECT.)

https://www.postgresql.org/docs/devel/spi-spi-connect.html